File: solutions/ruby/sieve/sieve.rb
class Sieve
  private

  attr_writer :primes

  def initialize(limit)
    primes = []
    sieve = Array.new(limit + 1, true)
    (2..limit).each do |number|
      next unless sieve[number]

      primes.append(number)
      (number..limit).step(number) { |mark| sieve[mark] = false }
    end
    self.primes = primes
  end

  public

  attr_reader :primes
end

File: solutions/python/sieve/sieve.py
def primes(limit: int) -> list[int]:
    limit += 1
    sieve = [True for i in range(limit)]
    for candidate in range(2, limit):
        if not sieve[candidate]:
            continue
        for not_prime in range(2 * candidate, limit, candidate):
            sieve[not_prime] = False
    return [candidate for candidate, is_prime in enumerate(sieve) if candidate > 1 and is_prime]
